graspologic/plot/plot.py

```diff
@@ -673,6 +676,223 @@ def pairplot(
+        input graph
+    gmm: Gaussian Mixture Model object
+    labels : 1d np.ndarray or list, same length as dimensions of X
+        Labels for different categories of graph nodes
+    label_palette : str or dict, optional,
+        with no input : dictionary using 'Set1'
+    cluster_palette : str or dict, optional, default: dictionary using 'Set1'
+        with no input : dictionary using 'Set1'
+    title : string
+        default : "Fit a Gaussian mixture using " + n_comps + " components"
+    context :  None, or one of {talk (default), paper, notebook, poster}
+        Seaborn plotting context
+    font_scale : float, optional, default: 1
+        Separate scaling factor to independently scale the size of the font
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    """
+    # Handle X
+    if not isinstance(X, (list, np.ndarray)):
+        msg = "X must be array-like, not {}.".format(type(X))
+        raise TypeError(msg)
+
+    # Handle labels
+    if labels is not None:
+        if not isinstance(labels, (list, np.ndarray)):
+            msg = "Y must be array-like or list, not {}.".format(type(labels))
+            raise TypeError(msg)
+        elif X.shape[0] != len(labels):
+            msg = "Expected length {}, but got length {} instead for Y.".format(
+                X.shape[0], len(labels)
+            )
+            raise ValueError(msg)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise NameError(msg)
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    data = pd.DataFrame(data=X)
+    num_comps = gmm.n_components
+    if labels is None:
+        lab_names = [i for i in range(num_comps)]
+        data["labels"] = np.asarray([lab_names[Y_[i]] for i in range(Y_.shape[0])])
+    else:
+        data["labels"] = labels
+    data["clusters"] = Y_
+    if isinstance(cluster_palette, str) and labels is not None:
+        colors = sns.color_palette(cluster_palette, num_comps)
+        cluster_palette = dict(zip(np.unique(Y_), colors))
+    if isinstance(label_palette, str) and labels is not None:
+        colors = sns.color_palette(label_palette, num_comps)
+        label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+    if (labels is not None) and (cluster_palette is None or label_palette is None):
+        msg = "if give labels array must give palette dicts or string desciptors"
+        raise ValueError(msg)
+    elif labels is None and cluster_palette is None and label_palette is None:
+        colors = sns.color_palette("Set1", num_comps)
+        cluster_palette = dict(zip(np.unique(lab_names), colors))
+        label_palette = dict(zip(np.unique(lab_names), colors))
+    elif labels is None and (cluster_palette is not None or label_palette is not None):
+        msg = "must give labels if using palettes for labels and clusters"
```

Comment:
format this a bit more and add comments? this code is really hairy to look at

Reply:
I meant to push the comments I had put in this morning!

Reply:
fixed https://github.com/microsoft/graspologic/pull/541/commits/0260d546589143489fa6e96d1aed686b329ae30b